tests/unit/test_awsproxy.py

```diff
@@ -663,6 +663,61 @@ async def test_request_whitelisted(mock_getenv, mock_fetch, mock_session):
     assert_http_response(mock_fetch, expected)
 
 
+@pytest.mark.asyncio
+@patch("tornado.httpclient.AsyncHTTPClient.fetch", new_callable=CoroutineMock)
+@patch("os.getenv")
+async def test_request_with_base_url(mock_getenv, mock_fetch, mock_session):
+    # Given
+    upstream_request = HTTPServerRequest(
+        method="HEAD",
+        uri="base-url/awsproxy/bucket-name-1",
```

Comment:
I copied an existing test, and changed the upstream request path to include a mock of a Jupyter notebook base url